File: C# Fundamentals/C# OOP Basics/13. Polymorphism/Ex/P03_WildFarm/Models/Animals/AbstractOnes/Mammal.cs
using System;

public abstract class Mammal : Animal
{
    protected const double MouseWeightIncreasePerFoodPiece = 0.10d;
    protected const double DogWeightIncreasePerFoodPiece = 0.40d;

    public Mammal(string name, double weight, string livingRegion) : base(name, weight)
    {
        this.LivingRegion = livingRegion;
    }

    public string LivingRegion { get; set; }

    public override string ToString()
    {
        return $"{this.GetType()} [{this.Name}, {this.Weight}, {this.LivingRegion}, {this.FoodEaten}]";
    }
}

File: Java/01.Advanced/02.OOP/11.Polymorphism/Ex/src/t_03_wild_farm/animals/Mammal.java
package t_03_wild_farm.animals;

import java.text.DecimalFormat;

public abstract class Mammal extends Animal {
    private final String livingRegion;

    public Mammal(String animalName, String animalType, double animalWeight, String livingRegion) {
        super(animalName, animalType, animalWeight);
        this.livingRegion = livingRegion;
    }

    protected String getLivingRegion() {
        return livingRegion;
    }

    @Override
    public String toString() {
        DecimalFormat df = new DecimalFormat("0.##");

        return String.format("%s[%s, %s, %s, %d]", this.getAnimalType(), this.getAnimalName(),
                df.format(this.getAnimalWeight()), this.livingRegion, this.getFoodEaten());
    }
}
